### DISCUSSION TITLE
Searching all VMs by a resourcepool

### DISCUSSION THREAD

**Discussion by lugerlee** [vmware/govmomi#3180] [Community Help] [answered]
How to collect all VMs by a resource pool and its child pools?
Any document is appreciated!

**Comment by atc0005** [comment, marked as answer]
@lugerlee Take a look here:

https://github.com/vmware/govmomi/blob/main/examples/virtualmachines/main.go

https://github.com/vmware/govmomi/blob/252e157f27b33d5cebeecdb21212e3f3fbe44039/examples/virtualmachines/main.go#L39-L42

If you replace `c.ServiceContent.RootFolder` with a ManagedObjectReference to the resource pool it should return the VirtualMachine values from the specified pool recursively.

**Comment by lugerlee** [comment]
Thanks for your help!

**Comment by lugerlee** [comment]
Thanks for your help!
